metrics/cpu_enabled.go

```diff
@@ -18,14 +18,19 @@
 
 package metrics
 
-import "github.com/elastic/gosigar"
+import "github.com/shirou/gopsutil/cpu"
 
 // ReadCPUStats retrieves the current CPU stats.
 func ReadCPUStats(stats *CPUStats) {
-	global := gosigar.Cpu{}
-	global.Get()
+	// passing false to request all cpu times
+	timeStats, err := cpu.Times(false)
```

Comment:
According to [the documentation](https://godoc.org/github.com/shirou/gopsutil/cpu#Times), 

> TimesStat contains the amounts of time the CPU has spent performing different kinds of work. Time units are in USER_HZ or Jiffies (typically hundredths of a second). It is based on linux /proc/stat file.

So you need to make sure that the units are the same as those returned by gosigar. I wanted to check the documentation and their website seems to be down. I suggest you compare both outputs.

Reply:
No they are not the same units. 

`Gosigar` uses a unit that is 128*(1 jiffy)

I just pushed a potential fix here: https://github.com/ethereum/go-ethereum/pull/21041/commits/397b1503e24a0b9b1dd2db8d31c38a17cf198e46

Reply:
Where does the 128 come from? Could yo upoint me to the gosigar code? I don't see any processing made by gosigar

Reply:
The code seems to contradict the documentation and indeed return times in seconds (in float, though :disappointed: )

In any case, it is unlikely that the 128 * jiffies is the same on all platforms. That's not looking too good if we can't get this to work with seconds :frowning_face:

Reply:
I would find it super odd if they returned anything that percentage or millisecond. That's the expectation .If they returned different on different platforms, nobody would use them. I think something's wrong with our assessment above

Reply:
@renaynay I created [this test program](https://github.com/gballet/test-gopsutil) and ran it on Linux like this:
```
> time ./test-gopsutil 
124820.68 28481.84 126.54 397.03 1.03505072e+06
12482068 2848184 12654 39703
./test-gopsutil  0.01s user 0.01s system 0% cpu 5.015 total
```

I do not find a difference of 128 but of 100 between the two. So if anything else, it proves that the 128 is platform-dependent. I'd be curious to find out what the output is when you run the same command on darwin, though.

Reply:
@karalabe They **do** return "CPU ticks" which is neither a percentage nor a millisecond, and they are platform dependent:

`parseCPUStats`, defined in https://github.com/elastic/gosigar/blob/master/sigar_linux.go#L95, parses a line taken from `/proc/stat` which returns values in "cpu ticks" and this is platform-dependent.

Reply:
Which makes me think: we can calculate the percentages. This would be accurate, whatever the unit is that is returned as they are divided by one another.